lake/sorted.go

```diff
@@ -46,19 +50,68 @@ func (s *statScanner) Pull() (zbuf.Batch, error) {
 	return batch, err
 }
 
+type rangeWrapper struct {
```

Comment:
Nit: This file will read better if you put `rangeWrapper` and methods after `newSortedScanner`.